### ISSUE TITLE
The Missing Link(s):  a disastrous oversight in `brew link`

### ISSUE THREAD

**Issue by gsteemso** [mistydemeo/tigerbrew#958]
Obviously, a keg-only formula does not get linked into /usr/local/bin.  That’s correct and expected behaviour.

HOWEVER, it turns out that such a formula also does not get linked into `${HOMEBREW_PREFIX}/opt` or `${HOMEBREW_REPOSITORY}/Library/LinkedKegs`!  If it, in theory, does get linked to those places, I have no idea what circumstances could reliably compel that to happen.  I’m also more than slightly baffled that only _some_ of my installed formulæ are missing these links.  Do different formulæ use different internal mechanisms to complete their installation?

This gross defect of the toolset has lost me over a week of scarce free time, chasing bizarre errors with no apparent cause.  When almost all of my installed formulæ are keg-only and over a quarter of them are not correctly linked, it causes *just a few problems!*

As there is no way to work out just what formulæ were compiled in the absence of these missing symlinks, I am faced with the daunting possibility of having to nuke the lot of them and reinstall *everything*.  Please tell me that there's a sanity-check, somewhere in the brewing system, that would cause a build failure instead of The Wrong Thing when one of these symlinks is absent!

Also, FEATURE REQUEST:  Some sort of rudimentary tool that can verify – and if need be, regenerate – these symlinks, for all installed packages.  Having to go through and laboriously re-link the dratted things by hand, one at a time, was an exercise in sheer misery.  At least I'm about 90% confident I didn’t muck any of them up with some subtle typo?

I'd like to contribute such a tool myself, as well as several minor bug fixes and code improvements I've been accumulating, but I'm still preoccupied with bootstrapping the Tigerbrew installation on my PPC G5 back to minimal functionality.  At the moment I'm fighting with OpenSSH, the last of the many dependencies of Git that have prevented me from installing it.  (Technically, I can't get cURL to build either, but I fiddled with the symlinks so it's been getting by on the self-contained version hidden in Tigerbrew's `vendor` directory.)

**Comment by gsteemso** [comment]
I figured out some time ago that formulæ ought to always be symlinked into `$PREFIX/opt` and only into `LinkedKegs` when they are not keg-only.  However, I still haven't decrypted enough of Tigerbrew's Byzantine internal structure to determine whether sanity checks of the nature I proposed exist, or if they don't, where they should go if I were to add them.  Absent further comment from someone who can make sense of Tigerbrew's guts, I will eventually have to give up and close this issue... but I'd really rather learn how to fix it instead.
